gensim/matutils.py

```diff
@@ -332,9 +343,15 @@ def unitvec(vec):
     Output will be in the same format as input (i.e., gensim vector=>gensim vector,
     or numpy array=>numpy array, scipy.sparse=>scipy.sparse).
     """
+    norm = isnorm(norm)
     if scipy.sparse.issparse(vec): # convert scipy.sparse to standard numpy array
         vec = vec.tocsr()
-        veclen = numpy.sqrt(numpy.sum(vec.data ** 2))
+        if norm == 'l0':
+            veclen = len(vec.data)
```

Comment:
Yep. Thanks.

On Mon, Apr 4, 2016 at 8:04 PM, Devashish Deshpande <
notifications@github.com> wrote:

> In gensim/matutils.py
> <https://github.com/piskvorky/gensim/pull/649#discussion_r58429753>:
>
> >      if scipy.sparse.issparse(vec): # convert scipy.sparse to standard numpy array
> >          vec = vec.tocsr()
> > -        veclen = numpy.sqrt(numpy.sum(vec.data ** 2))
> > +        if norm == 'l0':
> > +            veclen = len(vec.data)
>
> Thanks for the explanation! I think it would be best to remove L0 norm
> then and keep the normalization just for l1 and l2. Should I proceed with
> it?
>
> —
> You are receiving this because you were assigned.
> Reply to this email directly or view it on GitHub
> <https://github.com/piskvorky/gensim/pull/649/files/8b27eb4f46e0854fefdb4b55b6cea6c42c11a14d#r58429753>
>